src/spyglass/utils/dj_mixin.py

```diff
@@ -697,25 +706,103 @@ def super_delete(self, warn=True, *args, **kwargs):
+        """Hash upstream table keys for no transaction populate.
+
+        Uses a RestrGraph to capture all upstream tables, restrict them to
+        relevant entries, and hash the results. This is used to check if
+        upstream tables have changed during a no-transaction populate and avoid
+        the following data-integrity error:
+
+        1. User A starts no-transaction populate.
+        2. User B deletes and repopulates an upstream table, changing contents.
+        3. User A finishes populate, inserting data that is now invalid.
+
+        Parameters
+        ----------
+        keys : list
+            List of keys for populating table.
+        """
+        RestrGraph = self._graph_deps[1]
+
+        if not (parents := self.parents(as_objects=True, primary=True)):
+            raise RuntimeError("No upstream tables found for upstream hash.")
+
+        leaves = {  # Restriction on each primary parent
+            p.full_table_name: [
+                {k: v for k, v in key.items() if k in p.heading.names}
+                for key in keys
+            ]
+            for p in parents
+        }
+
+        return RestrGraph(seed_table=self, leaves=leaves, cascade=True).hash
+
     def populate(self, *restrictions, **kwargs):
-        """Populate table in parallel.
+        """Populate table in parallel, with or without transaction protection.
 
         Supersedes datajoint.table.Table.populate for classes with that
-        spawn processes in their make function
+        spawn processes in their make function and always use transactions.
+
+        `_use_transaction` class attribute can be set to False to disable
+        transaction protection for a table. This is not recommended for tables
+        with short processing times. A before-and-after hash check is performed
+        to ensure upstream tables have not changed during populate, and may
+        be a more time-consuming process. To permit the `make` to insert without
+        populate, set `_allow_insert` to True.
         """
-
-        # Pass through to super if not parallel in the make function or only a single process
         processes = kwargs.pop("processes", 1)
+
+        # Decide if using transaction protection
+        use_transact = kwargs.pop("use_transation", None)
+        if use_transact is None:  # if user does not specify, use class default
+            use_transact = self._use_transaction
+            if self._use_transaction is False:  # If class default is off, warn
+                logger.warning(
+                    "Turning off transaction protection this table by default. "
+                    + "Use use_transation=True to re-enable.\n"
+                    + "Read more about transactions:\n"
+                    + "https://docs.datajoint.io/python/definition/05-Transactions.html\n"
+                    + "https://github.com/LorenFrankLab/spyglass/issues/1030"
+                )
+        if use_transact is False and processes > 1:
```

Comment:
Why is transaction protection required with multiple processes? 

In the case of no-transaction multiprocesses could we:
- make the list of keys to populate
- make a pool of workers to run the make function 
- check the hash after

Might be worth it since this condition is specifically made for long-running make functions

Reply:
`populate` also reserves jobs. I was worried about the pool running duplicate work, but I didn't dive into allocator enough to know that that wouldn't be a concern. Do you know if folks are running multi-process for these non-transact tables?

Are you imagining a parallel `make_pass_function` that runs the same hash check before returning?

Reply:
I was thinking the loop in line 795 could be a multiprocessing pool `map` function mapping the list of keys to make.  Checks would still work the same and in the same place

Reply:
Yeah, I think I would kick the can pending either demand from our side or DJ's 'split-make' PR. I'd prefer a cautious rollout of anything that impacts `populate`